### DISCUSSION TITLE
Make a morphed Target Mesh Rigidbody

### DISCUSSION THREAD

**Discussion by techcoder-04** [enable3d/enable3d#111] [Help / Questions]
Hey thanks for the awesome work

I have a sphere mesh that morph Targets to 1 . Suzanne 2 . Torus and 3 . Square
After applying morph influences it gives a mixture of all the shapes 

I exported it into a GLTF model in three.js
Until this everything works fine

But I need to make this model into rigidbody

````
const loader = new THREE.GLTFLoader().load('/assets/glb/model.gltf', function(gltf) {
          console.log(gltf)
          const model= gltf.scene.children[0]
          model.position.setY(5)

          model.traverse(child => {
            if (child.isMesh) {
              child.castShadow = child.receiveShadow = false
              child.material.metalness = 0
              child.material.roughness = 1
            }
          })

    scene.add(model)
    physics.add.existing(model, { shape: 'mesh' })
})

````

When doing this it only adds physics to the old original mesh ( Sphere ) and not to the morphed part

Is there any way to provide entire morphed geometry and make it completely into a rigidbody
I had posted a copy in three.js forum too [https://discourse.threejs.org/t/get-the-geometry-or-vertices-of-morphedtarget-mesh/23459](https://discourse.threejs.org/t/get-the-geometry-or-vertices-of-morphedtarget-mesh/23459)
Hope anyone can help me with this .

**Comment by yandeu** [comment]
> Hey thanks for the awesome work

Thanks 😊



> After applying morph influences it gives a mixture of all the shapes

I don't know what _applying morph influences_ means, but if you cloud somehow share your model I could try to figure it out.

**Reply by techcoder-04** [reply]
Hey Thanks for the reply

Here is an example for the MorphTarget program in three.js 
[Example](https://threejs.org/examples/webgl_morphtargets.html)

And this is the source code for it
[Code](https://github.com/mrdoob/three.js/blob/dc2b340839e0fb1cf810a08dfbccf534121606b8/examples/webgl_morphtargets.html#L134)

My problem is When I export the morphed model [ In the above example a sphere and twisted morphed model ] and apply physics to it , It only applies physics to the original square , ignoring the morphs applied

_PS : I don't know if that's the problem only from my side - When I click on the links above it directs me to 404 page , If you too face that please visit the links seperately they work fine_

**Reply by yandeu** [reply]
> PS : I don't know if that's the problem only from my side - When I click on the links above it directs me to 404 page , If you too face that please visit the links seperately they work fine

Your links are wrong. You have to use the format:  
`[GitHub](http://github.com)`.

You do:  
`[http://github.com](link)`.

Please change it :)

**Reply by techcoder-04** [reply]
Thanks for the help . Changed the link format . Now it works fine :)

**Reply by yandeu** [reply]
@techcoder-04 Thanks for opening the issue https://github.com/mrdoob/three.js/issues/21287.

**Reply by techcoder-04** [reply]
Thanks for your effort on solving the issue 

Were you been able to figure out how to use computeMorphAttributes to make rigidbodies
I added [this fiddle](https://jsfiddle.net/techcoder04/rme37t9h/) in [the forum](https://discourse.threejs.org/t/get-the-geometry-or-vertices-of-morphedtarget-mesh/23459/13)

Can morphAttributes be used in enable3d to get the morphed geometry's rigidbody
